### DISCUSSION TITLE
Resolution Independence

### DISCUSSION THREAD

**Discussion by HardlineStudios** [mikke89/RmlUi#708] [Q&A]
Hello everyone, I had first posted this in the issue tracker which was the wrong spot. Sorry!

So my goal is to get the RmlUI library built for Windows, Linux and Android and then integrating it into my older game engine which still runs GL/GLES 2.0 and has to run on different platforms with vastly different aspect ratios and resolutions.

What is the best way to approach this? Is it just a matter of using % rather then px or dp for sizing inside .RML files? This seemed to work on the demo.rml file I tried it on at least. 

Cheers!

**Comment by mikke89** [comment]
Hey, welcome! And no worries.

In terms of dynamic layout, it sounds like you have discovered the most important things already. The `dp` unit is very useful here, and then you simply scale that relative to true pixels globally on the context. You might also find the `vw` and `vh` units useful.

We also support media queries, so you can specify things for e.g. different aspect ratios and resolutions too. Best of luck! Please let us know how things go, and I'm always interested to see the results of what you make with the library :)

**Reply by HardlineStudios** [reply]
Hmm ok. Well using % works but elements don't maintain their size ratios so that's another issue that needs to be solved.

Is this something I can do in the CSS/HTML code?

**Reply by mikke89** [reply]
I'm not sure I follow, maybe you could post an example?

**Reply by HardlineStudios** [reply]
My bad, I meant vw and vh but I think I'm getting the hang of it now!
